lib/attache/upload.rb

```diff
@@ -25,7 +25,14 @@ def _call(env, config)
           return [500, config.headers_with_cors.merge('X-Exception' => 'Local file failed'), []]
         end
 
-        config.async(:storage_create, relpath: relpath, cachekey: cachekey) if config.storage && config.bucket
+        if config.storage && config.bucket
+          request.body.rewind if request.body.respond_to?(:rewind)
+          if Attache.outbox.write(request_hostname(env), relpath, request.body)
```

Comment:
Looking into the return value of `Attache.outbox.write(request_hostname(env), relpath, request.body)`,
and so was looking at the return value of the final line in the `Attache.outbox.write`:

```
open(destpath, 'wb') { |dest| IO.copy_stream(src, dest) }
```

According to documentation, the return value of `open` is:

http://ruby-doc.org/core-2.2.0/File.html#method-c-open

> the value of the block will be returned from File.open.
 
And so, it will be thr return value of `IO.copy_stream(src, dest)`:

http://ruby-doc.org/core-2.2.0/IO.html#method-c-copy_stream

> This method returns the number of bytes copied.

Does this mean that `copy_stream` can essentially return `0 bytes` which means that nothing was copied and should be a failure?

Does `if Attache.outbox.write(request_hostname(env), relpath, request.body)` need to specifically check that it returns > 0?